[14.0][MIG] delivery_postlogistics to 14.0

hailangvn opened the pull request.

yvaucher committed:
[ADD] delivery_carrier_label_laposte module for Swiss Postlogistics Barcode WebService

yvaucher committed:
[ADD] external dependency on suds

yvaucher committed:
[IMP] module description

yvaucher committed:
[ADD] technical informations

yvaucher committed:
[IMP] rename module delivery_carrier_label_laposte => delivery_carrier_label_postlogistics

jgrandguillaume committed:
[FIX] When using update_postlogistics_options action, we must take the company contain in the current configuration (res.config) instead of the company of the user

yvaucher committed:
[IMP] adapt delivery_carrier_label_postlogistics to generate a list of label per picking tracking

[IMP] adapt to signature change in base_delivery_carrier_label for generate_pack_labels -> generate_shipping_labels

[IMP] labels on stock.picking instead of stock.picking.out so we can generate pretty PostLogistics labels even for the not yet happy customers who have to return product.

[ADD] pot file

Commit:
[ADD] missing access rights

[FIX] wrong method name

yvaucher committed:
[IMP] Permits to use multiple basic services

[IMP] postlogistics - licenses management

[FIX] wrong field name for license number

yvaucher committed:
[ADD] security groups for postlogistics licenses

[IMP] PEP8

bealdav committed:
[IMP] move postlogistic menu in submenu

Commit:
[ADD] use the new parameter 'tracking_ids' to generate only the labels of the trackings given in tracking_ids if used.

The implementation by default still generate them all at once from a button, but for instance the module allowing to print labels of a dispatch use it

use the name of the picking when there is no pack

yvaucher committed:
Consider returned sPDF file as PDF file. Those file are PDF without integrated font

[FIX] make sure pack number (which can be picking name) has no unallowed character for item id of postlogistics label

delivery_carrier_label_postlogistics - Improve carrier type selection inheritance

Update API version and when recipient partner's field street2 is defined we rewrite name and steet label address lines to print it on label.

Use AddressSuffix instead of reforging the address lines that way we don't break company name addition

Update PostLogistics webservice doc version

[FIX] fr.po file by removing ^M breaklines

[FIX] postlogistics warnings unwrapping

Improve file_type selection field inheritance

add extension to file name

Commit:
wrong docstring, trackings expects a list of browse record

yvaucher committed:
revert rev 35 for conflict resolution

Replay of Improve file_type selection field inheritance

yvaucher committed:
revert rev 35 for conflict resolution

Commit:
Launchpad automatic translations update.

yvaucher committed:
set installable to false

Commit:
Move delivery_carrier_label_postlogistics from __unported__ to root folder

Set delivery_carrier_label_postlogistics installable

Extract the description in README.rst

New API for res.company

Migration of delivery.py to the new API

The state field has been replaced by boolean fields

Migration of stock picking to the new API

And API changes in base_delivery_carrier_label

compute method must assign a recordset, not a list

Wrong assignation

Migration of res.config to new API

Fix a comment

Comma at end of line

Migrated base_delivery_carrier_label so adapt the call

Migrated here: https://github.com/bealdav/carrier-delivery/pull/1

The field on the package is now parcel_tracking

The relation is now package_id

The domain prevents other modules to show their options

The drawback of this change is that postlogistics will also
show the options of the others carriers but I can't find a way
to filter them out. The user still see in the search that an option
is not related to postlogistics.

Fallback on 'en' when no lang is provided

Update the carrier inside the loop

Get the allowed options on the parent instead of the related

The related allowed_option_ids is computed before carrier_id is set and is
incorrect. Also, a change needs to be saved to see the change back on the
related field.

Hide the option of other carriers such as colissimo

The domain in the view is insufficient because it displays the options with an
empty partner.  Adding dynamically the domain when carrier's partner is
postlogistics ensures that only postlogistics' options are shown.

Fix: W503 line break before binary operator

Readme more similar than the OCA's template

Add license key and OCA logo

Remove dead code

Implement Postlogistics Cash on Delivery labels

The webservice awaits the cash on delivery amount.
When a whole sales order is delivered in one picking / one package, we the
amount is the total of the sales order. If it is split in several pickings /
packages, we don't know the value of each package and so we ask the user to
fill the price on the packages.

Update translations

Fix error when user has no language

Set default language to 'en_US' when the user has no language

tafaRU committed:
Protect suds import with a try-except ImportError so that the presence of this module in the addons path will not cause an error even if the addon is not installed.

sbidoul committed:
[FIX] remove duplicate key in manifest

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

yvaucher committed:
[PORT] delivery_carrier_label_postlogistics port to 9.0
- Reactivate
- Move files in views and models
- Adapt XML views
- Update README.rst

[PORT] delivery_carrier_label_postlogistics - move license config in a separate view list. This fix an issue of write on related One2many in res.config.settings object

Update README.rts with links and requirements

Add missing set default on licenses

Use short license header

Set module category

<openerp> replaced by <odoo> tag in xml files

Update README.rst

Fix postlogistics tests

Commit:
Fix assignation on recordset with len > 1

Commit:
Rename field type to carrier_type in delivery.carrier

Missing acl for stock user/manager

pedrobaeza committed:
[MIG] Make modules uninstallable

[MIG] Rename manifest files

yvaucher committed:
Use configuration_helper to simplify the settings class (#84)

Add test mode using integration server of Swiss post + however, it needs https://fedorahosted.org/suds/attachment/ticket/239/suds_recursion.patch on suds library

Conflicts:
	delivery_carrier_label_postlogistics/company.py
	delivery_carrier_label_postlogistics/postlogistics/web_service.py
	delivery_carrier_label_postlogistics/res_config.py
	delivery_carrier_label_postlogistics/res_config_view.xml

Add tracking number format option.

By default let postlogistics generate the tracking number.

The new option is to use picking number or concatenation of picking
number and pack counter.

Conflicts:
	delivery_carrier_label_postlogistics/company.py
	delivery_carrier_label_postlogistics/postlogistics/web_service.py
	delivery_carrier_label_postlogistics/res_config.py
	delivery_carrier_label_postlogistics/res_config_view.xml

Implement delivery instructions ZAW

Improve error handling for 3 default parameters layout, output format and resolution

Refactor generation of tracking number to make it overridable

Don't send unset data for delivery instruction in order to get the error message from webservice

Add a note on patch to apply + a copy of it in the module

Add nocache with suds to force the reload of the definition otherwise it
work once then fail on next request.

Concatenate all pack tracking numbers on picking carrier tracking ref field when using packs

Add option to print ProClima logo

Fix assignation of tracking ref

Fix errors on check of parameters

yvaucher committed:
[MIG] delivery_carrier_label_postlogistics to 10.0

* Simplify signature of soap connector by removing company
* As carrier inherits from product, add a service postlogistics product
* Use an image instead of base64 for postlogistics logo

yvaucher committed:
Fix missing decorator on fields_view_get

Fix tests with change of xmlid

grindtildeath committed:
Fix domain on delivery.carrier.option tmpl_option_id_domain field. Use delivery_type instead of carrier_type.

Use partner_id instead of product to create template options

Define postlogistics_ methods

Add to contributors

Fix travis

Improvements after review

Define new readme folder for generator

Fix manifest, import, rst, etc

Automatically patch suds library on module loading

This allows to use suds library without having to patch manually for recursion issue.
Moreover it logs any HTTP error and displays it with a meaningful error.

grindtildeath committed:
[MIG] delivery_carrier_label_postlogistics : Migration to v11

grindtildeath committed:
Remove old wsdl and xsd files, change manual to english last version

grindtildeath committed:
Add default values for invoice_policy and expense_policy

grindtildeath committed:
Fix flake8 / pylint

sebalix committed:
[IMP] delivery_carrier_label_postlogistics: improve test coverage

sebalix committed:
[FIX] delivery_carrier_label_postlogistics: display Postlogistics fields correctly on the delivery carrier form

grindtildeath committed:
Improve tests with application of options on picking

grindtildeath committed:
[IMP] Consider parent partner name if partner name is empty

When using a specific partner contact (e.g. shipping address) as recipient,
the name of the partner can be empty. In such case, the name of its parent
must be used.

sebalix committed:
[MIG] delivery_carrier_label_postlogistics: Migration to 12.0

'default_' configuration fields have a special meaning for Odoo configuration
settings panel (to define default values on a given data model).
As such the following fields on 'res.company' have been renamed on purpose:

    - postlogistics_default_label_layout -> postlogistics_label_layout
    - postlogistics_default_output_format -> postlogistics_output_format
    - postlogistics_default_resolution -> postlogistics_resolution

jcoux committed:
[MIG] delivery_carrier_label_postlogistics: Migration to new API

yvaucher committed:
Unit tests for REST API

yvaucher committed:
Reactivate an occulted test

yvaucher committed:
Remove PDF content to not bloat the logs

yvaucher committed:
sanitize strings

Remove all non ascii char from labels as the API is unstable.

Curently:

- passing only a street with accents is ok
- passing only a name with accents is ok
- passing a name AND a street with accents is NOT ok

pedrobaeza committed:
[FIX] delivery_carrier_label_postlogistics: README RST syntax

santostelmo committed:
Fix access to ir.config_parameter

yvaucher committed:
Filter credentials in VCR.py cassettes

Those are integration only credentials

yvaucher committed:
Postlogistics - Recycle access token

The REST API have limits on the number of requests to access token.
To tackle this and generate hundred of labels the access token is
saved for later uses.

Made threadsafe on the token request to be usable with
deliver_carrier_label_batch module.

i-vyshnevska committed:
[12.0][IMP] generate labels for all packages in stock picking

simahawk committed:
delivery_carrier_label_postlogistics: make not installable

santostelmo committed:
delivery_postlogistics: add pack_weight parameter on _prepare_attributes

phuctranfxvn committed:
[REF] delivery_carrier_label_postlogistics becomes delivery_postlogistics in 13.0

sebalix committed:
delivery_postlogistics: avoid duplication of labels

mmequignon committed:
Do not send disallowed chars on postlogistics api

Some chars are disallowed from postlogistics API ("|", "", "<", ">")
and output json should be sanitized

mmequignon committed:
delivery_postlogistics: Refactor tests

sebalix committed:
delivery_postlogistics: several fixes

* fix 'PostLogistics' terms
* make 'Package Code' required for PostLogistics delivery packaging
* show a link listing available package codes for PostLogistics delivery
  packaging
* update README

hailangvn committed:
[MOV] change module name, prepare to change file structure

mmequignon committed:
delivery_postlogistics: Handle case where no label is returned by the API

hailangvn committed:
[MIG] delivery_postlogistics: Migration to 14.0

The cassette `tests/fixtures/cassettes/test_store_label.yaml` should be
updated as the label was failed to decode in both 13.0 and 14.0.

* https://github.com/odoo/odoo/blob/13.0/odoo/addons/base/models/ir_attachment.py#L212
* https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/ir_attachment.py#L554

We can remove the method _selection_file_type with below reasons.

* It is used in base_delivery_carrier_label but was removed on Sep 5, 2019
  https://github.com/OCA/delivery-carrier/blame/12.0/base_delivery_carrier_label/models/shipping_label.py#L15
* PR migration to 13.0 of base_delivery_carrier_label was merged on Oct 22, 2020
  https://github.com/OCA/delivery-carrier/pull/251/files
* Dependency on base_delivery_carrier_label was removed in migration to 14.0 on Apr 14 2021
  https://github.com/OCA/delivery-carrier/commit/4ba2b56e2e26ed2fd7b5b2f36b42b6f3060c45f7#diff-f16078e35000c05a9b63f5e7e04b6f5fd50eadae8b14a45aa9c1eb078031060cL12 https://github.com/OCA/delivery-carrier/commit/4ba2b56e2e26ed2fd7b5b2f36b42b6f3060c45f7#diff-08c11b571d9c0b0c509c3922c77f407ef4c429473fda9bc2d4d542520e7a4e3fR12

leemannd commented on delivery_postlogistics/views/delivery.xml:
```diff
@@ -0,0 +1,95 @@
+            </page>
+        </field>
+    </record>
+    <!-- <record
```
Missing migration or to be removed?

hailangvn replied:
I will remove this code in next update.

leemannd commented on delivery_postlogistics/models/stock_quant_package.py:
```diff
@@ -0,0 +1,77 @@
+                )
+            )
+
+        # As history_ids column is removed, and stock.quant is not directly
```
Missing migration of code or to be removed?

hailangvn replied:
There is comment that we may need a feature to have multiple packages for single delivery order. I am looking for suggestion.

mmequignon committed:
[13.0] delivery_postlogistics: Add disallowed chars

joao-p-marques committed:
[IMP] *: pre-commit execution

hailangvn commented:
Good day, this PR has been updated with improvements from 13.0.

mmequignon commented:
runbot is :tomato:

nilshamerlinck commented:
Impacted by https://github.com/OCA/maintainer-quality-tools/pull/697

nilshamerlinck commented:
Certificate issue mentioned above has been fixed. We are now looking into the other issue that prevents runbot from being green.

hailangvn commented:
All is green now. :)

simahawk commented on delivery_postlogistics/__manifest__.py:
```diff
@@ -3,15 +3,21 @@
     "license": "AGPL-3",
     "category": "Delivery",
     "complexity": "normal",
+    "external_dependencies": {
```
this is wrong. It's not a fucntional dependency. Such dep goes to test-requirements.txt at repo level.

simahawk commented on delivery_postlogistics/models/stock_quant_package.py:
```diff
@@ -12,17 +12,20 @@ class StockQuantPackage(models.Model):
         move_line_model = self.env["stock.move.line"]
-        move_line = move_line_model.search([("package_id", "=", self.id)])
-        return move_line.mapped("picking_id")
+        move_line = move_line_model.search([("result_package_id", "=", self.id)])
```
This change seems wrong. The method says "_get_origin" but here you look for destination packages now :thinking:

sebalix commented on delivery_postlogistics/models/stock_picking.py:
```diff
@@ -0,0 +1,237 @@
+            )
+        order_moves = order.picking_ids
+        # check if the package delivers the whole sales order
+        if len(order_moves) > 1:
```
```suggestion
        # check if the package delivers the whole sales order
        if len(order.picking_ids) > 1:
```

sebalix commented on delivery_postlogistics/models/stock_quant_package.py:
```diff
@@ -12,17 +12,20 @@ class StockQuantPackage(models.Model):
         move_line_model = self.env["stock.move.line"]
         move_line = move_line_model.search([("package_id", "=", self.id)])
-        return move_line.mapped("picking_id")
+        return move_line.mapped("picking_id.sale_id.picking_ids")
```
Question: is this change part of the migration or a bug you spotted? If it's a bug, better to put such change in a separate commit so they can be backported easily on previous versions.

sebalix commented on delivery_postlogistics/models/stock_quant_package.py:
```diff
@@ -59,16 +62,4 @@ def postlogistics_cod_amount(self):
-                    "in several packages."
-                )
-            )
-
```
What is the reason to remove this?

sebalix replied:
<s>Comment outdated, resolved!</s> I think my comment is still relevant, the migration commit is removing this piece of code.

hailangvn replied:
Above fix and this one are split and combined into new commit.

sebalix reviewed (commented):
Some questions

hailangvn committed:
[FIX] delivery_postlogistics: remove obsoleted code, get sale pickings

history_ids has been removed as below link. So the code is obsoleted.

https://github.com/odoo/odoo/commit/b3180c841101510081ee8ef9c52d205497efdd4f#diff-9a22420c4a72cc9b6d558b3d832846c2e44a4f3731c6df9579d66c6279c4804eL54-L57

_get_origin_pickings is to determine number of pickings of a sale order.
So it should get pickings from sale_id of found picking.

hailangvn commented:
Thanks @sebalix for your review. This PR has been updated accordingly.

simahawk commented:
@sebalix good for you?

sebalix reviewed (approved):
Yes, LG
@mmequignon will probably fix the tests/cassettes as some changes occurred on SwissPost API, let's merge this already.

simahawk commented:
/ocabot merge nobump
